pom.xml

```diff
@@ -112,14 +112,34 @@
                         </configuration>
                     </execution>
                     <execution>
-                        <id>gulp build</id>
-                        <goals>
-                            <goal>gulp</goal>
-                        </goals>
-                        <phase>generate-resources</phase>
-                        <configuration>
-                            <arguments>build</arguments>
-                        </configuration>
+                      <id>gulp brand</id>
+                      <goals>
+                        <goal>gulp</goal>
+                      </goals>
+
+                      <!-- optional: the default phase is "generate-resources" -->
+                      <phase>generate-resources</phase>
+
+                      <configuration>
+                        <!-- optional: if not specified, it will run gulp's default
+                        task (and you can remove this whole <configuration> section.) -->
+                        <arguments>brand --repo ${branding.repo} --branch ${branding.branch}</arguments>
```

Comment:
Can we not just have this one execution as the default? We could define branding.rep and branding.branch in the maven repo and then if we want to pull from a different repo/branch we just specify `mvn clean install -Dbranding.repo=blah`

Reply:
Ok, it's a deal

Reply:
Yeah, let's but at the moment we don't have those branches. It is easy to add them as properties in pom.xml for both products. We yet have to agree on how we are going to do that. In the meantime, I made changes so that branding only occurs if repo and branch are specified. Until then then build stays essentially the same. I'll commit that change now

Reply:
So at the moment when we call `mvn clean install`, both `gulp brand` and then `gulp build` are being called. We should just need to call `gulp build`. In `gulpfile.js` we can simply add `brand` as one of the requirements for the `build` task. If no `cmdParams.repo` or `cmdParams.branch` is specified, then do nothing (the current code in `brand.js`). In the pom.xml, we can set the default value of `branding.repo` and `branding.branch` to be empty, so unless these parameters are set no branding with external repos occur when `mvn clean install` is called.

Reply:
Ok +1